### PR TITLE
chore: warn if the action is pinned to HEAD

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -35,10 +36,28 @@ const oidcWarning =
   `GitHub Actions workflow permissions are incorrect, or this job is being ` +
   `run from a fork. For more information, please see https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token`;
 
+const headWarning =
+  `google-github-actions/auth is pinned at HEAD. We strongly advise against ` +
+  `pinning to "@main" as it may be unstable. Please update your GitHub ` +
+  `Action YAML from:\n` +
+  `\n` +
+  `    uses: 'google-github-actions/auth@master'\n` +
```
Comment (bharathkkb) [inline src/main.ts:44]:
```suggestion
  `    uses: 'google-github-actions/auth@main'\n` +
```

**Review by bharathkkb — COMMENTED** [review]
LGTM

File: src/main.ts [hunk 2]
```diff
@@ -35,10 +36,28 @@ const oidcWarning =
   `GitHub Actions workflow permissions are incorrect, or this job is being ` +
   `run from a fork. For more information, please see https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token`;
 
+const headWarning =
```
Comment (bharathkkb) [inline src/main.ts:39]: This could be something we can put into utils too so we can just have this message in a single place.

Comment (sethvargo) [reply src/main.ts:39]: https://github.com/google-github-actions/actions-utils/pull/2

**Pull request merged**